traits/testing/unittest_tools.py

```diff
@@ -352,13 +352,16 @@ def assertMultiTraitChanges(
             The extended trait names of traits not expected to change.
 
         """
-        args = []
-        for obj in objects:
-            for trait in traits_modified:
-                args.append(self.assertTraitChanges(obj, trait))
-            for trait in traits_not_modified:
-                args.append(self.assertTraitDoesNotChange(obj, trait))
-        return _py2to3.nested_context_mgrs(*args)
+        with contextlib.ExitStack() as exit_stack:
```

Comment:
Note that there's a subtle behaviour change here. The old code instantiates all the context managers first, and *then* passes them off to be entered, one-by-one. The new code enters each context manager immediately after creating it, so the pattern is now [init, enter, init, enter, init, enter] rather than [init, init, init, enter, enter, enter].

It doesn't make a difference in this case, but more generally the new pattern is preferable: the old pattern doesn't clean up properly in cases where resources are allocated at init time rather than enter time (as with opening files, for example). This is exactly the reason that contextlib.nested was deprecated and removed. The removed nested_context_mgrs code deliberately reproduced the old, flawed behaviour of contextlib.nested.